components/support/ffi/src/lib.rs

```diff
@@ -343,3 +343,19 @@ fn init_backtraces_once() {
 
 #[cfg(not(feature = "log_backtraces"))]
 fn init_backtraces_once() {}
+
+#[repr(C)]
+pub struct ProtoWrapper {
+    pub data: *mut u8,
```

Comment:
These can't be pub, if they are then you can cause memory unsafety from purely safe rust. Alternatively, `destroy()` should be unsafe. Possibly both.

If you make them non-pub, be sure to document the order they appear in, since while it's not `pub`, it is part of the public API. See how we do it on ExternError.